pkg/test/spaceprovisionerconfig/spaceprovisionerconfig.go

```diff
@@ -24,6 +25,38 @@ func NewSpaceProvisionerConfig(name string, namespace string, opts ...CreateOpti
 	return spc
 }
 
+func NewEnabledValidSpaceProvisionerConfigWithTenantRole(name string, namespace string, referencedToolchainCluster string, opts ...CreateOption) *toolchainv1alpha1.SpaceProvisionerConfig {
```

Comment:
minor thing - but TBH I find it easier to read the content of the function, then reading the long name. 

My question is - what is the use case/benefit for having those `New..` constructor functions, couldn't we just avoid adding those permutations and just rely on `NewSpaceProvisionerConfig` with the needed options?

Reply:
I'm also not a huge fan of long names but at the same time, this function is used on *85* places (and the "overrides" of it on some more), so I thought it would be good to have 1 way that consistently sets up the SPC the same way everywhere rather specifying the 4 options on every one of those 80+ places.

I wanted to be explicit about what the function does instead of calling it "NewInitializedSpc" or something.

Reply:
how about `NewEnabledValidTenantSPC`?